neuralop/layers/spectral_convolution.py

```diff
@@ -470,45 +447,32 @@ def forward(
         starts = [(size - min(size, n_mode)) for (size, n_mode) in zip(list(x.shape[2:]), list(weight.shape[weight_start_idx:]))]
         slices_x =  [slice(None), slice(None)] # Batch_size, channels
 
-        if self.complex_data:
+        if self.complex:
             slices_x += [slice(start//2, -start//2) if start else slice(start, None) for start in starts]
         else:
             slices_x += [slice(start//2, -start//2) if start else slice(start, None) for start in starts[:-1]]
             slices_x += [slice(None, -starts[-1]) if starts[-1] else slice(None)] # The last mode already has redundant half removed
         out_fft[slices_x] = self._contract(x[slices_x], weight, separable=self.separable)
 
-        if self.output_scaling_factor is not None and output_shape is None:
-            mode_sizes = tuple([round(s * r) for (s, r) in zip(mode_sizes, self.output_scaling_factor[indices])])
+        if self.resolution_scaling_factor is not None and output_shape is None:
+            mode_sizes = tuple([round(s * r) for (s, r) in zip(mode_sizes, self.resolution_scaling_factor)])
 
         if output_shape is not None:
             mode_sizes = output_shape
 
         if self.order > 1:
             out_fft = torch.fft.fftshift(out_fft, dim=fft_dims[:-1])
         
-        if self.complex_data:
+        if self.complex:
             x = torch.fft.ifftn(out_fft, s=mode_sizes, dim=fft_dims, norm=self.fft_norm)
         else:
             x = torch.fft.irfftn(out_fft, s=mode_sizes, dim=fft_dims, norm=self.fft_norm)
 
         if self.bias is not None:
-            x = x + self.bias[indices, ...]
+            x = x + self.bias
 
         return x
 
-    def get_conv(self, indices):
-        """Returns a sub-convolutional layer from the joint parametrize main-convolution
-
-        The parametrization of sub-convolutional layers is shared with the main one.
-        """
-        if self.n_layers == 1:
-            Warning("A single convolution is parametrized, directly use the main class.")
-
-        return SubConv(self, indices)
-
-    def __getitem__(self, indices):
-        return self.get_conv(indices)
-
 
 class SubConv(nn.Module):
```

Comment:
I don't think we need SubConv anymore, I only wrote those for supporting Multiple Layers parameterized jointly

Reply:
Agreed and removed